caddyhttp/httpserver/server.go

```diff
@@ -490,6 +490,20 @@ func DefaultErrorFunc(w http.ResponseWriter, r *http.Request, status int) {
 	WriteTextResponse(w, status, fmt.Sprintf("%d %s\n", status, http.StatusText(status)))
 }
 
+const errStatusMisdirectedRequest = 421 // RFC 7540, 9.1.2
+// WriteSiteNotFound writes appropriate error code to w, signaling that
+// requested host is not served by Caddy on a given port.
+func WriteSiteNotFound(w http.ResponseWriter, r *http.Request) {
+	var status int
+	if r.ProtoMajor >= 2 {
+		// TODO: use http.StatusMisdirectedRequest when it gets defined
+		status = errStatusMisdirectedRequest
+	} else {
+		status = http.StatusNotFound
+	}
+	WriteTextResponse(w, status, fmt.Sprintf("%d No site %s served at this port\n", status, r.Host))
```

Comment:
For clarity, rephrase to `Site %s is not served on this interface` - if that sounds good to you?